File: kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/quickstart/ProductClient.java
package io.micronaut.configuration.kafka.docs.quickstart;

// tag::imports[]
import io.micronaut.configuration.kafka.annotation.KafkaClient;
import io.micronaut.configuration.kafka.annotation.KafkaKey;
import io.micronaut.configuration.kafka.annotation.Topic;
// end::imports[]

// tag::clazz[]
@KafkaClient // <1>
public interface ProductClient {

    @Topic("my-products") // <2>
    void sendProduct(@KafkaKey String brand, String name); // <3>

    void sendProduct(@Topic String topic, @KafkaKey String brand, String name); // <4>
}
// end::clazz[]

File: test-suite-python/src/test/python/micronaut/kafka/docs/quickstart/ProductClient.py
from abc import ABC, abstractmethod
from typing import Annotated

# tag::imports[]
from micronaut.configuration.kafka.annotation import KafkaClient, KafkaKey, Topic
from micronaut.context.annotation import Requires
# end::imports[]


@Requires(property="spec.name", value="QuickstartTest")
# tag::clazz[]
@KafkaClient  # <1>
class ProductClient(ABC):

    @Topic("my-products")  # <2>
    @abstractmethod
    def send_product(self, brand: Annotated[str, KafkaKey], name: str) -> None:  # <3>
        ...

    @abstractmethod
    def send_product_to_topic(self, topic: Annotated[str, Topic], brand: Annotated[str, KafkaKey], name: str) -> None:  # <4>
        ...
# end::clazz[]
